demo/src/main/java/com/demo/model/Book.java

```diff
@@ -0,0 +1,31 @@
+package com.demo.model;
+
+import jakarta.annotation.Nonnull;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@Entity
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Nonnull
```

Comment:
you should enforce non-nullability on DB level for JPA entities. use column(nullable=false)

Reply:
thank you, refactored